source/ed/ed_reader.cpp

```diff
@@ -729,30 +731,35 @@ void EdReader::fill_vehicle_journeys(nt::Data& data, pqxx::work& work){
             mvj_name = vj_name;
         }
         auto mvj = data.pt_data->meta_vjs.get_or_create(mvj_name);
-        std::string rt_level_str;
-        const_it["vj_class"].to(rt_level_str);
         auto rt_level = navitia::type::get_rt_level_from_string(const_it["vj_class"].as<std::string>());
+        const auto& vp = *validity_pattern_map[const_it["validity_pattern_id"].as<idx_t>()];
+        const auto uri = const_it["uri"].as<std::string>();
+        const auto vj_id = const_it["id"].as<idx_t>();
         if (const_it["is_frequency"].as<bool>()) {
-            auto f_vj = mvj->create_frequency_vj(rt_level);
-            route->frequency_vehicle_journey_list.push_back(f_vj);
+            auto f_vj = mvj->create_frequency_vj(uri,
+                                                 rt_level,
+                                                 vp,
+                                                 route,
+                                                 std::move(sts_from_vj[vj_id]),
```

Comment:
I imagine that `vj_id` is supposed to be existent in `sts_from_vj`, any chance that it's not the case?